packages/api/src/router/widgets/stocks.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from "zod";
+
+import { fetchStockPriceHandler } from "@homarr/request-handler/stock-price";
+
+import { stockPriceTimeFrames } from "../../../../widgets/src/stocks";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+const stockPriceInputSchema = z.object({
+  stock: z.string().nonempty(),
+  timeRange: z.enum([...stockPriceTimeFrames.range]),
+  timeInterval: z.enum([...stockPriceTimeFrames.interval]),
+});
+
+export const stockPriceRouter = createTRPCRouter({
+  getPrices: publicProcedure.input(stockPriceInputSchema).query(async ({ input }) => {
```

Comment:
I first did not understand why you used plural. Maybe we could rename it to `getPriceHistory`